sqlbrite/src/androidTest/java/com/squareup/sqlbrite/SqlBriteContentProviderTests.java

```diff
@@ -0,0 +1,191 @@
+  private static final String KEY = "test_key";
+  private static final String VALUE = "test_value";
+
+  private final RecordingObserver o = new RecordingObserver();
+
+  private SqlBriteContentProvider db;
+  private Subscription subscription;
+
+  public SqlBriteContentProviderTests() {
+    super(TestContentProvider.class, AUTHORITY.getAuthority());
+  }
+
+  @Override protected void setUp() throws Exception {
+    super.setUp();
+    db = SqlBriteContentProvider.create(getMockContentResolver());
+    getProvider().init(getContext().getContentResolver());
+    db.setLoggingEnabled(true);
+    subscription = Subscriptions.empty();
+  }
+
+  @Override public void tearDown() {
+    o.assertNoMoreEvents();
+    subscription.unsubscribe();
+  }
+
+  public void testLoggerInvalidValues() {
+    try {
+      db.setLogger(null);
+      fail();
+    } catch (NullPointerException e) {
+      assertThat(e).hasMessage("logger == null");
+    }
+  }
+
+  public void testLoggerEnabled() {
+    final List<String> logs = new ArrayList<>();
+    db.setLogger(new SqlBrite.Logger() {
+      @Override public void log(String message) {
+        logs.add(message);
+      }
+    });
+
+    subscription = db.createQuery(TABLE, null, null, null, null, false).subscribe(o);
+    o.assertCursor().isExhausted();
+
+    getMockContentResolver().insert(TABLE, values("key1", "value1"));
+    o.assertCursor().hasRow("key1", "value1").isExhausted();
+    assertThat(logs).isNotEmpty();
+  }
+
+  public void testLoggerDisabled() {
+    final List<String> logs = new ArrayList<>();
+    db.setLoggingEnabled(false);
+    db.setLogger(new SqlBrite.Logger() {
+      @Override public void log(String message) {
+        logs.add(message);
+      }
+    });
+
+    getMockContentResolver().insert(TABLE, values("key1", "value1"));
+    assertThat(logs).isEmpty();
+  }
+
+  public void testCreateQueryObservesInsert() {
+    subscription = db.createQuery(TABLE, null, null, null, null, false).subscribe(o);
+    o.assertCursor().isExhausted();
+
+    getMockContentResolver().insert(TABLE, values("key1", "val1"));
+    o.assertCursor().hasRow("key1", "val1").isExhausted();
+  }
+
+  public void testCreateQueryObservesUpdate() {
+    getMockContentResolver().insert(TABLE, values("key1", "val1"));
+    subscription = db.createQuery(TABLE, null, null, null, null, false).subscribe(o);
+    o.assertCursor().hasRow("key1", "val1").isExhausted();
+
+    getMockContentResolver().update(TABLE, values("key1", "val2"), null, null);
+    o.assertCursor().hasRow("key1", "val2").isExhausted();
+  }
+
+  public void testCreateQueryObservesDelete() {
+    getMockContentResolver().insert(TABLE, values("key1", "val1"));
+    subscription = db.createQuery(TABLE, null, null, null, null, false).subscribe(o);
+    o.assertCursor().hasRow("key1", "val1").isExhausted();
+
+    getMockContentResolver().delete(TABLE, null, null);
```

Comment:
Can we put `getMockContentResolver()` in a field? It's referenced one or more times in every test.